### PR TITLE
Finalize docs for new System.Net.Primitives API-s

### PR DESCRIPTION
**Pull request by antonfirsov**
## Summary

Port/add all remaining doc changes for `System.Net.Primitives` finishing the work started in #9169. I added most  `<inheritdoc/>` bits manually, since `PortToDocs` didn't really work well here.

Contributes to https://github.com/dotnet/runtime/issues/88569.

/cc @gewarren @carlossanlop

### CODE REVIEW

**Comment by antonfirsov** [comment]
Converting to a draft, since `<inheritdoc />` doesn't work the way I expected.

**Comment by antonfirsov** [comment]
@gewarren this is ready for review now.

File: xml/System.Net/IPAddress.xml [hunk 1]
```diff
@@ -2085,12 +2085,10 @@
         <ReturnType>System.Net.IPAddress</ReturnType>
       </ReturnValue>
       <Parameters>
-        <Parameter Name="ipString" Type="System.ReadOnlySpan&lt;System.Char&gt;" Index="0" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
```
Comment (gewarren) [inline xml/System.Net/IPAddress.xml:2088]: I think we need to keep this for .NET Core 2.1 - 3.1.

File: xml/System.Net/IPAddress.xml [hunk 2]
```diff
@@ -2085,12 +2085,10 @@
...
         <Parameter Name="ipSpan" Type="System.ReadOnlySpan&lt;System.Char&gt;" Index="0" FrameworkAlternate="net-5.0;net-6.0;net-7.0;net-8.0" />
       </Parameters>
       <Docs>
-        <param name="ipString">A character span that contains an IP address in dotted-quad notation for IPv4 and in colon-hexadecimal notation for IPv6.</param>
```
Comment (gewarren) [inline xml/System.Net/IPAddress.xml:2092]: Keep ipString here too.

File: xml/System.Net/IPAddress.xml [hunk 3]
```diff
@@ -2561,10 +2559,10 @@
         <Parameter Name="bytesWritten" Type="System.Int32" RefType="out" Index="1" FrameworkAlternate="net-8.0" />
       </Parameters>
       <Docs>
-        <param name="utf8Destination">To be added.</param>
-        <param name="bytesWritten">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="utf8Destination">When this method returns, the IP address as a span of UTF-8 bytes.</param>
```
Comment (gewarren) [inline xml/System.Net/IPAddress.xml:2562]:
```suggestion
        <param name="utf8Destination">The span into which to write the IP address as a span of UTF-8 bytes.</param>
```

Comment (gewarren) [reply xml/System.Net/IPAddress.xml:2562]: The "When this method returns" text is specific to out parameters.

File: xml/System.Net/IPAddress.xml [hunk 4]
```diff
@@ -2759,7 +2757,6 @@
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Parameters>
-        <Parameter Name="ipString" Type="System.ReadOnlySpan&lt;System.Char&gt;" Index="0" FrameworkAlternate="netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1;netcore-3.1" />
```
Comment (gewarren) [inline xml/System.Net/IPAddress.xml:2762]: Keep.

File: xml/System.Net/IPAddress.xml [hunk 5]
```diff
@@ -2775,12 +2772,11 @@
         </Parameter>
       </Parameters>
       <Docs>
-        <param name="ipString">The byte span to validate.</param>
```
Comment (gewarren) [inline xml/System.Net/IPAddress.xml:2778]: Keep both ipString and ipSpan.

File: xml/System.Net/IPNetwork.xml [hunk 6]
```diff
@@ -164,6 +171,7 @@
         <returns>
           <see langword="true" /> if the current object is equal to the <paramref name="other" /> parameter; otherwise, <see langword="false" />.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.InvalidOperationException">Uninitialized <see cref="T:System.Net.IPNetwork" /> instance.</exception>
```
Comment (gewarren) [inline xml/System.Net/IPNetwork.xml:174]:
```suggestion
        <exception cref="T:System.InvalidOperationException">The <see cref="T:System.Net.IPNetwork" /> instance is uninitialized.</exception>
```

File: xml/System.Net/IPNetwork.xml [hunk 7]
```diff
@@ -198,10 +206,12 @@
...
       <Docs>
-        <param name="obj">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="obj">The <see cref="T:System.Net.IPNetwork" /> instance to compare to this instance.</param>
+        <summary>Determines whether two <see cref="T:System.Net.IPNetwork" /> instances are equal.</summary>
+        <returns>
+          <see langword="true" /> if <paramref name="obj" /> is an <see cref="T:System.Net.IPNetwork" /> instance and the networks are equal; otherwise <see langword="false" />.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.InvalidOperationException">Uninitialized <see cref="T:System.Net.IPNetwork" /> instance.</exception>
```
Comment (gewarren) [inline xml/System.Net/IPNetwork.xml:214]:
```suggestion
        <exception cref="T:System.InvalidOperationException">The <see cref="T:System.Net.IPNetwork" /> instance is uninitialized.</exception>
```

**Review by gewarren — APPROVED** [review]

File: xml/System.Net/IPAddress.xml (hunk 1 above)
Comment (antonfirsov) [reply xml/System.Net/IPAddress.xml:2088]: But that belongs to the other overload, just above this one:
```
<Member MemberName="Parse">
      <MemberSignature Language="C#" Value="public static System.Net.IPAddress Parse (string ipString);" />
```

This overload has only the `ipSpan` parameter. Or is there some interdependence between the overloads docs? How does that work?

Comment (gewarren) [reply xml/System.Net/IPAddress.xml:2088]: It looks like the parameter was renamed in .NET 5. The XML keeps both versions of the method signature. You can see the old one here:

image

**Review by carlossanlop — APPROVED** [review]

**Pull request merged**